### DISCUSSION TITLE
Keyball46 firmware v1.0.0 RC1

### DISCUSSION THREAD

**Discussion by koron** [Yowkees/keyball#31] [Show and tell]
これよりも新しいファームが https://github.com/Yowkees/keyball/discussions/39 に置いてあります。

---

Keyball46 の v1.0.0 のリリース候補(RC1)のビルド済みファームウェアを用意しました。

* [keyball_via_1.0.0-rc.1.zip](https://github.com/Yowkees/keyball/files/6721521/keyball_via_1.0.0-rc.1.zip) - VIA対応、右トラックボール
* [keyball_via_Left_1.0.0-rc.1b.zip](https://github.com/Yowkees/keyball/files/6721553/keyball_via_Left_1.0.0-rc.1b.zip) - VIA対応、左トラックボール
* [keyball_test_1.0.0-rc.1.zip](https://github.com/Yowkees/keyball/files/6721520/keyball_test_1.0.0-rc.1.zip) - 組み立て時のテスト用、右トラックボール
    RGB LEDがテストパターンで(時間で赤→緑→青に)点灯する
* [keyball_test_Left_1.0.0-rc.1.zip](https://github.com/Yowkees/keyball/files/6721580/keyball_test_Left_1.0.0-rc.1.zip) - 組み立て時のテスト用、左トラックボール
* [keyball_default_1.0.0-rc.1.zip](https://github.com/Yowkees/keyball/files/6721519/keyball_default_1.0.0-rc.1.zip) - ミニマルなファームウェア、トラックボール側にUSBを接続すること
    OLEDおよびRGB LEDに非対応

このバージョンのファームウェアではトラックボールの有無に応じて
基板ごとに書き込むファームウェアが異なります。
各ZIPには `_rev1_ball_` と `_rev1_noball_` の2つのファームが含まれています。
`_rev1_ball_` をトラックボールのある方に、
`_rev1_noball_` をトラックボールのない方に書き込んでください。

例: VIA対応の中身

* `keyball_rev1_ball_via.hex` - トラックボールのあるPCB用
* `keyball_rev1_noball_via.hex` - トラックボールのないPCB用

**Comment by koron** [comment]
via_Left版の PRODUCT_ID を `0x0002` に変更した版 (rc.1b) です。

[keyball_via_Left_1.0.0-rc.1b.zip](https://github.com/Yowkees/keyball/files/6721553/keyball_via_Left_1.0.0-rc.1b.zip)

See https://github.com/Yowkees/keyball/issues/30

**Reply by koron** [reply]
念のため test_Left も追加しました。

**Comment by Yowkees** [comment]
うわーーーすごいスピード感！きっと今日の販売開始に合わせていただいたんだと思います。ありがとうございます！
テストファームを入れたのですが、すごい楽しいです。このボールの読み取り値はくるくるしてて楽しくなるので、ぜひ標準ファームにも搭載したくなります。
また、メモリーオーバー期に削除してしまったのですが、最後に入力されたキーの表示も追加できると右手OLEDが賑やかになりそうです。
すでに最終ファームをアップしていただいた後ですから、お時間あるときで結構ですのでご検討いただければ幸いです。

**Comment by koron** [comment]
これよりも新しいファームが https://github.com/Yowkees/keyball/discussions/39 に置いてあります。
